### PR TITLE
Modules cleanup, May 2023

### PR DESCRIPTION
**Pull request by aaime**
As per our previous discussion on gs-devel.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by jodygarnett** [comment]
Q: Should we update the [notes on specific versions](https://docs.geoserver.org/latest/en/user/installation/upgrade.html#notes-on-upgrading-specific-versions) when extensions reach EOL? Or just better to have the extension vanish without a trace ... pop!

**Comment by aaime** [comment]
> Q: Should we update the [notes on specific versions](https://docs.geoserver.org/latest/en/user/installation/upgrade.html#notes-on-upgrading-specific-versions) when extensions reach EOL? Or just better to have the extension vanish without a trace ... pop!

We have tickets for each module, so it will be a surprise for the few that read that document but do not care reading the release notes and/or the release announcement. So far module movement has been traced only with Jira tickets. Want to propose changing this on the devel list?

**Comment by bradh** [comment]
> > Q: Should we update the [notes on specific versions](https://docs.geoserver.org/latest/en/user/installation/upgrade.html#notes-on-upgrading-specific-versions) when extensions reach EOL? Or just better to have the extension vanish without a trace ... pop!
> 
> We have tickets for each module, so it will be a surprise for the few that read that document but do not care reading the release notes and/or the release announcement. So far module movement has been traced only with Jira tickets. Want to propose changing this on the devel list?

An blog post before the release would soften the blow. Might result in clamoring for "save my mission critical unsupported community module", but gives people a chance to find a support mechanism if they are really that serious.

File: doc/en/user/source/installation/docker.rst [hunk 1]
```diff
@@ -173,26 +173,26 @@ For the current list see GeoServer `build server <https://build.geoserver.org/ge
...
+    features-templating        netcdf-ghrsst              smart-data-loader           
+    flatgeobuf                 notification               solr                        
+    gdal-wcs                   ogcapi-coverages           spatialjson                 
+    gdal-wps                   ogcapi-dggs                stac-datastore              
+    geopkg                     ogcapi-features            taskmanager-core            
+    gpx                        ogcapi-images              taskmanager-s3              
+    gsr                        ogcapi-maps                vector-mosaic               
+    gwc-azure-blobstore        ogcapi-styles              vsi                         
+    gwc-distributed            ogcapi-tiled-features      webp                        
+    gwc-mbtiles                ogcapi-tiles               wms-eo
```
Comment (bradh) [inline doc/en/user/source/installation/docker.rst:193]:
```suggestion
    gwc-mbtiles                ogcapi-tiles 
```

File: doc/en/user/source/installation/docker.rst [hunk 2]
```diff
@@ -173,26 +173,26 @@ For the current list see GeoServer `build server <https://build.geoserver.org/ge
...
-    gwc-azure-blobstore        ogcapi-maps                teradata                    
-    gwc-distributed            ogcapi-styles              vector-mosaic               
-    gwc-mbtiles                ogcapi-tiled-features      vsi                         
-    gwc-sqlite                 ogcapi-tiles               webp                        
-    hz-cluster                 ogr-datastore              wms-eo                      
-    importer-jdbc              opensearch-eo              wmts-styles                 
-    jdbc-metrics               ows-simulate               wps-remote                  
+    activeMQ-broker            jdbcconfig                 proxy-base-ext           
+    backup-restore             jdbcstore                  s3-geotiff                  
+    cog                        jms-cluster                saml
```
Comment (bradh) [inline doc/en/user/source/installation/docker.rst:178]: Probably need to remove `saml`

Comment (aaime) [reply doc/en/user/source/installation/docker.rst:178]: Yep

**Comment by aaime** [comment]
Mail sent asking for feedback on users and devel list, none received. Merging.

**Pull request merged**